services/network/virtualnetwork/virtualnetwork.go

```diff
@@ -136,7 +136,13 @@ func getWssdNetworkSubnets(subnets *[]network.Subnet) (wssdsubnets []*wssdcloudn
 		}
 
 		if subnet.NetworkSecurityGroup != nil {
-			wssdsubnet.NetworkSecurityGroup = *subnet.NetworkSecurityGroup.ID
+			wssdsubnet.NetworkSecurityGroup = &wssdcommonproto.ResourceReference{
+				Name: *subnet.NetworkSecurityGroup.ID,
+			}
+
+			if location != nil {
+				wssdsubnet.NetworkSecurityGroup.Location = *location
```

Comment:
>wssdsubnet.NetworkSecurityGroup.Location = *location [](http://example.com/codeflow?start=4&length=52)

qq, this is now optional right?

Reply:
yes! I had it not optional for a second, but fixed it. The wssd definition doesn't really have a way to mark a field as optional, but I'm only consistently filling in the name